### PR TITLE
feat: dissallow names not between 2 and 32 characters long #49 

### LINKED ISSUE #49: Dissallow names less than 2 characters long or more than 32 characters

### PR DESCRIPTION
**Pull request by SichangiFelix**
Here are some screenshots of the functionality.

This screenshot shows behavior with less than two characters

name less than 2

This screenshot shows behavior with name having more than 2 and less than 32  characters

correct number of characters

This screenshot shows behavior with more than 32characters

name more than 32

### CODE REVIEW

**Review by EnokiUN — APPROVED** [review]

File: lib/routes/home.dart [hunk 1]
```diff
@@ -1,10 +1,35 @@
 import 'package:eludris/routes/loggedin.dart';
 import 'package:flutter/material.dart';
 
-class Home extends StatelessWidget {
+class Home extends StatefulWidget {
+  Home({Key? key}) : super(key: key);
```
Comment (teaishealthy) [inline lib/routes/home.dart]:
```suggestion
  const Home({Key? key}) : super(key: key);
```

File: lib/routes/home.dart [hunk 2]
```diff
@@ -1,10 +1,35 @@
...
+
+  //Method to validate input
+  validateNameInput(String input) {
+    if (input.trim().length < 2 || input.trim().length > 32) {
+      setState(() {
+        buttonEnabled = false;
+      });
+      ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+          content: Text(
+              'Ensure the name is more than 2 and less than 32 characters long')));
```
Comment (teaishealthy) [inline lib/routes/home.dart]:
```suggestion
              'Username must be between 2 and 32 characters long')));
```

**Review by teaishealthy — CHANGES_REQUESTED** [review]
Currently if the user keeps typing after 32 characters multiple Snackbars will be shown.
I would replace the Snackbar with hint text under the input would as that is less annoying.

I'd love to hear other suggestions and ideas if you got any!

File: lib/routes/home.dart (hunk 1 above)
Comment (SichangiFelix) [reply lib/routes/home.dart]: Yeah that sounds better, let me do that

**Review by SichangiFelix — COMMENTED** [review]
I have made the requested changes

**Review by teaishealthy — CHANGES_REQUESTED** [review]
- State updates even where there is no new state
- The Widget being hidden causes the button to shift, try using Opacity?

**Comment by teaishealthy** [comment]
@SichangiFelix Any updates?

**Comment by SichangiFelix** [comment]
> 

I can have a look at the raised issue...

**Comment by teaishealthy** [comment]
Closing due to no updates

**Pull request closed without merging**